### PR TITLE
Add setting to skip smallest bloom pyramid levels

### PR DESCRIPTION
**Pull request by skerys**
### Purpose of this PR
This PR adds the setting to skip the smallest bloom pyramid levels. The number of pyramid levels was causing non-negligible driver overhead on mobile devices, skipping the smallest ones should increase performance while barely or not at all changing the effect.

---
### Testing status

**Manual Tests**: 
Ran bloom related graphics tests locally
PseudoCI A/B performance testing

**Automated Tests**: 
Universal graphics tests on iPhone_Metal, Win_DX11_Standalone, Android_Vulkan and Android_OpenGLES3

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/liutauras%252Fskip-bloom-iterations

**Any test projects to go with this to help reviewers?**:
Bloom related scenes from UniversalTestProject (059 and 100)

---

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/Overrides/Bloom.cs [hunk 1]
```diff
@@ -23,6 +23,13 @@ public sealed class Bloom : VolumeComponent, IPostProcessComponent
         [Tooltip("Use bicubic sampling instead of bilinear sampling for the upsampling passes. This is slightly more expensive but helps getting smoother visuals.")]
         public BoolParameter highQualityFiltering = new BoolParameter(false);
 
+        [Tooltip("Skip the smallest bloom pyramid steps.")]
+#if UNITY_ANDROID || UNITY_IOS
+        public ClampedIntParameter skipIterations = new ClampedIntParameter(2, 0, 16);
+#else
+        public ClampedIntParameter skipIterations = new ClampedIntParameter(0, 0, 16);
+#endif
```
Comment (Chman) [inline com.unity.render-pipelines.universal/Runtime/Overrides/Bloom.cs:31]: How does this behave when the user starts with let say a desktop project and changes the player to iOS later on?

Comment (skerys) [reply com.unity.render-pipelines.universal/Runtime/Overrides/Bloom.cs:31]: Currently, the default value is set only when creating the bloom override, so in your example, the skip count would still be at 0. Should that be changed? And if so, what would the expected behavior when changing build targets after changing the value (f.e. change skipIterations to 4 on a desktop project and then change the build target to iOS)

Comment (Chman) [reply com.unity.render-pipelines.universal/Runtime/Overrides/Bloom.cs:31]: That's the question I had, yes. I'm not sure what the expected behavior should be, but I would refrain from changing user values dynamically, that would be too unpredictable and annoying. I'd say either we just set the default to 2 no matter what the platform is, or we leave it at 0 and rely on documentation (risky move).
Overall it's better to have a predictable behavior than one that "tries to do magic" behind the users' back.

**Comment by ellioman** [comment]
Looks good. I'm adding @oleks-k to review tooltip and because docs probably need updating as well.

File: com.unity.render-pipelines.universal/Documentation~/post-processing-bloom.md [hunk 2]
```diff
@@ -27,6 +27,7 @@ To add **Bloom** to a Volume:
 | **Tint**                   | Use the color picker to select a color for the Bloom effect to tint to. |
 | **Clamp**                  | Set the maximum intensity that Unity uses to calculate Bloom. If pixels in your Scene are more intense than this, URP renders them at their current intensity, but uses this intensity value for the purposes of Bloom calculations. The default value is 65472. |
 | **High Quality Filtering** | Enable this to use high quality sampling. This reduces flickering and improves the overall smoothness, but is more resource-intensive and can affect performance. |
+| **Skip Iterations** | Set how many smallest bloom pyramid iterations to skip. This reduces driver overhead and increases performance, especially on mobile. The default value is 2. |
```
Comment (oleks-k) [inline com.unity.render-pipelines.universal/Documentation~/post-processing-bloom.md:30]: The last iterations in the processing sequence might have little contribution to the appearance of the rendered frame. In this setting you define the number of final iterations to skip. Increasing this value reduces processing load and increases performance, especially on mobile devices. The default value is 2.

File: com.unity.render-pipelines.universal/Runtime/Overrides/Bloom.cs [hunk 3]
```diff
@@ -23,6 +23,9 @@ public sealed class Bloom : VolumeComponent, IPostProcessComponent
         [Tooltip("Use bicubic sampling instead of bilinear sampling for the upsampling passes. This is slightly more expensive but helps getting smoother visuals.")]
         public BoolParameter highQualityFiltering = new BoolParameter(false);
 
+        [Tooltip("Skip the smallest bloom pyramid steps.")]
```
Comment (oleks-k) [inline com.unity.render-pipelines.universal/Runtime/Overrides/Bloom.cs:26]: The number of final iterations to skip in the effect processing sequence.

**Comment by ellioman** [comment]
Yamato was green before the final commit which was adding to the changelog.

**Pull request merged**